spark/src/main/scala/geotrellis/spark/io/s3/S3LayerDeleter.scala

```diff
@@ -0,0 +1,40 @@
+package geotrellis.spark.io.s3
+
+import com.amazonaws.services.s3.model.DeleteObjectsRequest.KeyVersion
+import com.amazonaws.services.s3.model.ObjectListing
+import geotrellis.spark.LayerId
+import geotrellis.spark.io._
+import spray.json.JsonFormat
+import spray.json.DefaultJsonProtocol._
+import scala.collection.JavaConversions._
+
+class S3LayerDeleter(attributeStore: AttributeStore[JsonFormat]) extends LayerDeleter[LayerId] {
+
+  def getS3Client: () => S3Client = () => S3Client.default
+
+  def deleteListing(s3: S3Client, bucket: String, listing: ObjectListing): Unit = {
```

Comment:
This looks a method that should be on the S3Client. Could use `tailrec` tag to make certain can't blow the stack.